api_request.go

```diff
@@ -39,40 +38,52 @@ func newDeadlineTransport(timeout time.Duration) *http.Transport {
 	return transport
 }
 
-func apiRequestNegotiateV1(method string, endpoint string, body io.Reader) (*simplejson.Json, error) {
+type generalResp struct {
+	Status     string      `json:"status_txt"`
+	StatusCode int         `json:"statuc_code"`
+	Data       interface{} `json:"data"`
+}
+
+func apiRequestNegotiateV1(method string, endpoint string, body io.Reader, ret interface{}) error {
 	httpclient := &http.Client{Transport: newDeadlineTransport(2 * time.Second)}
 	req, err := http.NewRequest(method, endpoint, body)
 	if err != nil {
-		return nil, err
+		return err
 	}
 
 	req.Header.Add("Accept", "application/vnd.nsq; version=1.0")
 
 	resp, err := httpclient.Do(req)
 	if err != nil {
-		return nil, err
+		return err
 	}
 
 	respBody, err := ioutil.ReadAll(resp.Body)
 	resp.Body.Close()
 	if err != nil {
-		return nil, err
+		return err
 	}
+
 	if resp.StatusCode != 200 {
-		return nil, fmt.Errorf("got response %s %q", resp.Status, respBody)
+		return fmt.Errorf("got response %s %q", resp.Status, respBody)
 	}
 
 	if len(respBody) == 0 {
 		respBody = []byte("{}")
 	}
 
-	data, err := simplejson.NewJson(respBody)
-	if err != nil {
-		return nil, err
+	if resp.Header.Get("X-NSQ-Content-Type") == "nsq; version=1.0" {
+		return json.Unmarshal(respBody, ret)
 	}
 
-	if resp.Header.Get("X-NSQ-Content-Type") == "nsq; version=1.0" {
-		return data, nil
+	gResp := &generalResp{
+		Data: ret,
```

Comment:
Yes :)
A `interface{}` value contains a type value and a pointer value.
So It actually set the value to the memory where `ret` address.
http://play.golang.org/p/Xqu5YY8e2L